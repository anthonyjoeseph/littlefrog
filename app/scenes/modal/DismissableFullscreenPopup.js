// app/DismissableFullscreenPopup.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TouchableHighlight,
  TouchableWithoutFeedback,
  Animated
} from 'react-native';

class DismissableFullscreenPopup extends Component {
  constructor(props) {
    super(props)
    this.isDisimissing = false;
    // set state with passed in props
    var {width, height} = Dimensions.get('window');
    this.state = {
      dimensions: {
        width: width,
        height: height
      },
      fadeAnim: new Animated.Value(0)
    }
    // bind functions
    this.dismiss = this.dismiss.bind(this)
  }

  componentDidMount() {
    Animated.timing(
      this.state.fadeAnim,
      {
        toValue: 1,
        duration: 300
      }
    ).start();
  }

  dismiss() {
    if(!this.isDisimissing){
      Animated.timing(
        this.state.fadeAnim,
        {
          toValue: 0,
          duration: 300
        }
      ).start(() => {
        this.props.onDismiss();
      });
      this.isDisimissing = true;
    }
  }

  // show or hide Modal based on 'hide' prop
  render() {
    return (
      <Animated.View style={[styles.outermostView, {
        opacity: this.state.fadeAnim,
        width: this.state.dimensions.width,
        height: this.state.dimensions.height
      }]}>
        <TouchableHighlight
          style={styles.touchableBackground}
          underlayColor='rgba(0, 0, 0, 0.0)'
          onPress={this.dismiss}>
          <View style={styles.transluscentBackground}>
            <TouchableWithoutFeedback>
              {this.props.children}
            </TouchableWithoutFeedback>
          </View>
        </TouchableHighlight>
      </Animated.View>
    )
  }
}

const styles = StyleSheet.create({
  outermostView:{
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
    position:'absolute',
    left:0, top:0,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'stretch'
  },
  touchableBackground:{
    flex:1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'stretch'
  },
  transluscentBackground:{
    flex:1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default DismissableFullscreenPopup;
