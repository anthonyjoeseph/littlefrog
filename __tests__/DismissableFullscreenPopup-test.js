// __tests__/DismissableFullscreenPopup-test.js
import {View, Text} from 'react-native';
import React from 'react';
import DismissableFullscreenPopup from '../app/scenes/modal/DismissableFullscreenPopup';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <DismissableFullscreenPopup
      onDismiss={()=>{}}
    >
      <View>
        <Text>Testing testing 123</Text>
      </View>
    </DismissableFullscreenPopup>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
