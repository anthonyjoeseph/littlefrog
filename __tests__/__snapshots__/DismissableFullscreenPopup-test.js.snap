exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "flexDirection": "row",
      "height": 1334,
      "justifyContent": "center",
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": 750,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={undefined}
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Testing testing 123
        </Text>
      </View>
    </View>
  </View>
</View>
`;
